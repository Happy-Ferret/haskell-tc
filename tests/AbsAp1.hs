module AbsAp1 where

data Maybe a = Just a | Nothing

id x = x

fromJust :: Maybe a -> a
fromJust mb =
  case mb of
    Just a -> id a


