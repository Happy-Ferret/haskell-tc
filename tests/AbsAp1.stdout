Bindings:
:: âˆ€ v2. v2 â†’  v2
  [4mid x = x[0m
:: v2
  id [4mx[0m = x
:: âˆ€ sk_3_a. Maybe sk_3_a â†’  sk_3_a
  [4mfromJust mb =[0m
  [4m  case mb of[0m
  [4m    Just a -> id a[0m
:: Maybe sk_3_a
  fromJust [4mmb[0m =
:: sk_3_a
      Just [4ma[0m -> id a

Proofs:
coercion: (âˆ€ v2. v2 â†’  v2) @ sk_3_a
      Just a -> [4mid[0m a
